t/full-stack.t

```diff
@@ -160,6 +160,11 @@ sub bail_with_log ($job_id, $message) {
     # uncoverable statement count:2
     my $log = eval { $log_file->slurp };
     diag $@ ? "unable to read $log_file: $@" : "$log_file:\n$log";    # uncoverable statement
+    my $worker_log_file = path("$resultdir/../pool/1/worker-log.txt");
```

Comment:
This line should still have an uncoverable statement but not about the sub. I suggest you read about the meaning of the different commands as it's good to know for other cases as well

Reply:
Why is it not failing then?

Reply:
I was assuming `uncoverable subroutine` was for `autoinst_log`; but it seems odd that this line is covered when the rest of the function isn't

Reply:
I added it back ^^

Reply:
> Why is it not failing then?

Hm, interesting. https://app.circleci.com/pipelines/github/os-autoinst/openQA/8420/workflows/87065537-ed48-42ab-bee7-ec7ec818274a/jobs/79591 shows that the full stack test succeeded on the first run. That statement is only for coverage collection. But why is https://app.codecov.io/gh/os-autoinst/openQA/compare/4363/diff?src=pr&el=tree#diff-dC9mdWxsLXN0YWNrLnQ= saying that the line is actually covered? I thought this function would only be called in case of failures?

Reply:
> But why is https://app.codecov.io/gh/os-autoinst/openQA/compare/4363/diff?src=pr&el=tree#diff-dC9mdWxsLXN0YWNrLnQ= saying that the line is actually covered?

Because Devel::Cover reports it as covered. Why? Because we say it's uncoverable, and then marking it as covered in the report which is sent to codecov is the only way to tell codecov that this line is uncoverable.

Reply:
I now tend to think it would be best to move all those uncoverable helper functions into a separate utility module which is then excluded completely from the report.